### DISCUSSION TITLE
Conflicting dependencies

### DISCUSSION THREAD

**Discussion by warri93** [renovatebot/renovate#26917] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Renovate creates a pull request to update my versions related to Angular. In my monorepo I also have storybook as a dependency. Storybook relies on @angular-devkit/architect, which should have a version between 0.1400.0 and 0.1700.0.

Within my Angular related pull request, @angular-devkit/architect gets updated to version 0.1701.1 which is out of range for storybook resulting in a conflict. Therefor my package-lock file can't be generated and my pull request is red. I do not want to allow conflicting dependency version by adding legacy-peer-deps=true in my .npmrc. 

Is there any configuration I can add in my renovate.json file to make sure the versions are updated based on every range it can find in my package-lock file to prevent this?

### Logs (if relevant)

<details><summary>Logs</summary>

```

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @storybook/angular@7.5.1
npm ERR! Found: @angular-devkit/architect@0.1701.1
npm ERR! node_modules/@angular-devkit/architect
npm ERR!   dev @angular-devkit/architect@"0.1701.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular-devkit/architect@">=0.1400.0 < 0.1700.0" from @storybook/angular@7.5.1
npm ERR! node_modules/@storybook/angular
npm ERR!   dev @storybook/angular@"7.5.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @angular-devkit/architect@0.1602.12
npm ERR! node_modules/@angular-devkit/architect
npm ERR!   peer @angular-devkit/architect@">=0.1400.0 < 0.1700.0" from @storybook/angular@7.5.1
npm ERR!   node_modules/@storybook/angular
npm ERR!     dev @storybook/angular@"7.5.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /tmp/renovate/cache/others/npm/_logs/2024-01-26T11_48_46_373Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /tmp/renovate/cache/others/npm/_logs/2024-01-26T11_48_46_373Z-debug-0.log


```

</details>

**Comment by rarkins** [comment]
The best thing you can do is group together related dependencies, such as storybook and the angular ones above
